packages/bot-skeleton/src/scratch/utils/index.js

```diff
@@ -587,9 +587,41 @@ export const isDarkRgbColour = string_rgb => {
 };
 /* eslint-enable */
 
-export const removeExtraInput = instance => {
+export const modifyMainBlockOnCollapse = block_instance => {
+    try {
+        // Return if the block is not collapsed
+        if (!block_instance?.collapsed_) return;
+        const [block_image, block_name] = block_instance?.inputList[0]?.fieldRow.map(field => field.value_);
+        const type_of_block = block_instance?.getField(block_instance.type);
+        if (type_of_block) return;
```

Comment:
```suggestion
        const type_of_block = block_instance?.getField(block_instance.type);
        if (type_of_block) return;
        
       const [block_image, block_name] = block_instance?.inputList[0]?.fieldRow.map(field => field.value_);
```

Let's change the sequence 🙏